File: kotlin-spring/src/main/kotlin/traveltimeplatform/models/ResponseRoute.kt
package traveltimeplatform.models

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import traveltimeplatform.models.ResponseRoutePart
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * 
 * @param departureTime 
 * @param arrivalTime 
 * @param parts 
 */
data class ResponseRoute (

        @get:NotNull 
        @JsonProperty("departure_time") val departureTime: java.time.OffsetDateTime,

        @get:NotNull 
        @JsonProperty("arrival_time") val arrivalTime: java.time.OffsetDateTime,

        @get:NotNull 
        @JsonProperty("parts") val parts: List<ResponseRoutePart>
) {

}

File: php-laravel/lib/app/Models/ResponseRoute.php
<?php
/**
 * ResponseRoute
 */
namespace app\Models;

/**
 * ResponseRoute
 */
class ResponseRoute {

    /** @var \DateTime $departureTime */
    private $departureTime;

    /** @var \DateTime $arrivalTime */
    private $arrivalTime;

    /** @var \app\Models\ResponseRoutePart[] $parts */
    private $parts;

}
